### ISSUE TITLE
nosupport

### ISSUE THREAD

**Issue by Apokleos** [kata-containers/packaging#313]

**Comment by grahamwhaley** [comment]
Hi @Apokleos - by default Kata containers guest kernels do not have module loading support enabled. It has been a long on-going discussion if we should have module loading enabled or not, and then that brings in more difficulties like where does the container find the modules to load etc. (they are not in the rootfs by default etc.).
There is a very recent issue discussing some of these items over at https://github.com/kata-containers/packaging/issues/298 that may help.

The quickest route for you to get your driver loaded into your container is going to be using the https://github.com/kata-containers/osbuilder repository to build yourself a customer kernel and possibly rootfs image with your driver enabled by default. If you can do that as a driver statically linked to the kernel then that will be easier, otherwise you will have to enable module loading to the kernel, add your module to the rootfs, and possibly work out/debug/add a module load/probe command.

**Comment by jodh-intel** [comment]
Also, there is some detail on kernels in https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md.

**Comment by grahamwhaley** [comment]
Actually, whilst doing some other stuff over the last couple of days I noticed:
- Our kernel config files look like they do have [module support enabled by default right now](https://github.com/kata-containers/packaging/blob/master/kernel/configs/x86_64_kata_kvm_4.14.x#L330)
- the osbuilder `rootfs.sh` looks like it has support to [add modules into the image already](https://github.com/kata-containers/osbuilder/blob/master/rootfs-builder/rootfs.sh#L95)

Those facts should hopefully ease your path @Apokleos :-)

**Comment by Apokleos** [comment]
Yeah, thanks very much for your (@grahamwhaley @jodh-intel ) suggestions. But I want to know more that if kata supports insmod/modprobe actions or not?
I am doing work which makes host device passed through kata containers. and the host device has passed through now. I want to make it work with its driver xxx.
First, I made a guest kernel with offical config https://github.com/kata-containers/packaging/blob/master/kernel/configs/x86_64_kata_kvm_4.14.x.
Second,  I made a docker build image with driver xxx source code copyed into docker image centos:7.6, and kernel/kernel-devel-kernel-headers-4.14.x installed.(makes sure that kernel in kata is 4.14.x )
Third, exec cmd: docker run -it --runtime=kata-runtime --rm --device /dev/vfio/48 -v /dev:/dev centos:7.6 /bin/bash
and the kata will be OK. But when I tried to insmod xxx.ko or modprobe xxx, it doesn't work. 
my steps as below:
0. I have compiled driver xxx, xxx.ko existed(it will work ok in normal VM), 
1. insmod xxx.ko
insmod: ERROR: could not insert module xxx.ko: Operation not permitted
2.modprobe xxx
modprobe: FATAL: Module xxx not found.
3. I will use 'dmesg', found nothing about insmod/modprobe actions.
By the way, we are trying  your  suggestions. 
Thx again.

**Comment by grahamwhaley** [comment]
@jodh-intel totally agree, this is an excellent candidate (and section) for the user guide. And, yes, having a label would make it easier to re-locate things that need harvesting .... my only worry would be that people might just slap on a label and avoid having to write the document(s) themselves... so we could end up with a lot of labels and no actual docs....
We can give it a try though. Let me make a label here and slap it on...

**Comment by jodh-intel** [comment]
> my only worry would be that people might just slap on a label and avoid having to write the document(s) themselves.

I thought that too. Maybe we could trawl the repos say weekly/monthly for the label and then have a mini-doc sprint or something where we can collaboratively work on the BBOK? Once the content is in the guide, we can remove the labels.

**Comment by jodh-intel** [comment]
Thanks @grahamwhaley. I've now added the same label to the other repos.
